rust-semantics/rust-common-syntax.md

```diff
@@ -540,7 +545,12 @@ https://doc.rust-lang.org/reference/items/extern-crates.html
 
 ```k
 
-  syntax StructExpression ::= "TODO: not needed yet, not implementing"
+  syntax StructExpression ::= TypePath "{" StructFieldsExpression "}"
```

Comment:
This does not follow the Rust reference (see the link at line 544). We should not follow it precisely when it helps us, but could you document the changes (a TODO would probably be enough)? Also, whenever possible, could you try to preserve the sort names in the Rust reference?

Some more precise questions:

1. Why did you use a TypePath instead of a PathInExpression here? If a TypePath is easier to implement, it's probably fine, since we don't plan to use generics in this way.
2. Does the `LiteralExpressionList` thing below actually work in Rust? By looking at this page, I'm not sure how: https://doc.rust-lang.org/reference/expressions/struct-expr.html . If it does not work in Rust, do we need it?
3. Does it help if you use LiteralExpression in `StructExprField` / `StructFieldAssignment` instead of Expression? The same question for `LiteralExpressionList` - you may be able to use `TupleElementsNoEndComma` here, which can be transformed to ExpressionList. I'm asking this because, if you use Expression/ExpressionList instead, you may have easier ways to evaluate expressions.

Reply:
1. - For the TypePath, I'm having issues converting a PathExpression into a TypePath, so I decided to use it directly in the construction. If you think the effort is valid, I can take some time to continue digging into it.
2. and 3. - In the first implementation, I've focused on implementing a syntax that would work for that specific case I was trying to handle on our tests. I've ended up not correcting the syntax once the test was being correctly executed, so I reimplemented the syntax to become closer to the reference implementation. Please notice that, even in this updated syntax, a few of the approaches that we use to initialize structs are not yet supported, and we're using a `TypePath` where a `PathInExpression` should be.